provider/pkg/provider/provider.go

```diff
@@ -1911,6 +1911,13 @@ func (k *kubeProvider) Create(
 		initialized = partialErr.Object()
 	}
 
+	// We need to delete the empty status field returned from the API server if we are in
+	// preview mode. Having the status field set will cause a panic during preview if the Pulumi
+	// program attempts to read the status field.
+	if req.GetPreview() {
+		unstructured.RemoveNestedField(initialized.Object, "status")
+	}
```

Comment:
What is the marshaled value of status? I ask because `plugin.MarshalProperties` is called with `SkipNulls: true`, which implies to me that this shouldn't be necessary. Since it is necessary, does `SkipNulls: true` not work?

Reply:
The `status` field is typically not nullable in Kubernetes objects, so I would expect that the `status` is empty, not null, here.

@rquitales could you elaborate on where the panic comes from? Is it in the application code or in the engine? 
`const ip = svc.status.apply((s) => s.loadBalancer.ingress[0].ip);`

Reply:
I'm almost certain that the panic comes from `s` being `null` (or `undefined`) in the lambda that is run during preview, instead of `svc.status` being unknown and the apply not running.

Reply:
Yes, the status field returned from a SSA dry run is empty. The panic comes from the application code: `const ip = svc.status.apply((s) => s.loadBalancer.ingress[0].ip)` since it's attempting to grab the IP address.

Without this fix, the status field is populated with an empty, but initialized object. So attempting to access `ingress[0]` will cause a panic. If we nullify the status field, then the panic is averted.